extensions/wikia/AdEngine/js/AdEngine2.js

```diff
@@ -68,11 +80,17 @@ define('ext.wikia.adEngine.adEngine', [
 					provider = providerList.shift();
 
 					if (!provider) {
-						return fillInSlotUsingProvider(slotname, adProviderNull, noop);
+						delete slot.success;
+
+						if (slot.error) {
```

Comment:
Yup, this check is stronger in a scenario when error is truthy and not a and function (will trigger js uncaught error) so if it's more valuable to fail early and explicitly than preserve graceful experience it should definitely stay this way :)